app/models/provider.rb

```diff
@@ -246,4 +252,15 @@ def set_defaults
     self.scheme_member ||= "is_a_UCAS_ITT_member"
     self.year_code ||= recruitment_cycle.year
   end
+
+  def services
+    return @services if @services.present?
+
+    @services = Dry::Container.new
+    @services.register(:generate_unique_course_code) do
+      Courses::GenerateUniqueCourseCodeService.new(
+        generate_course_code_service: Courses::GenerateCourseCodeService.new,
```

Comment:
Should we be planning to simplify this service by moving `GenerateUniqueCourseCodeService` (either it's initialisation or it's functionality) into `GenerateCourseCodeService`? Happy for that to be done at a later time, just wondering what we think about it's existence.

Reply:
We should do it at a later time I think - reduces the change area in this PR

Reply:
We should do it at a later time I think - reduces the change area in this PR. Though the reasoning behind it was to make sure we could test the logic of ensuring it was unique separate from the randomness (moving the randomness to a separate class)

If we wanna flatten it, we should try make sure the randomness is consistent between tests to ensure that behaviour is still tested